tools/flaky-test-reporter/slack_notification.go

```diff
@@ -0,0 +1,180 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// slack_notification.go sends notifications to slack channels
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"strings"
+	"sync"
+	"time"
+
+	"net/http"
+	"net/url"
+
+	"github.com/knative/test-infra/shared/testgrid"
+)
+
+const (
+	knativeBotName          = "Knative Testgrid Robot"
+	slackChatPostMessageURL = "https://slack.com/api/chat.postMessage"
+	// default filter for testgrid link
+	testgridFilter = "exclude-non-failed-tests=20"
+)
+
+var (
+	// jobNameTestgridURLMap contains harded coded mapping of job name: Testgrid tab URL relative to base URL
+	jobNameTestgridURLMap = map[string]string{
```

Comment:
maybe this var should be in testgrid.go and not here?

Reply:
I thought about it too. The problem of putting it there, is that it's under `shared`, so just put this one line mapping there would feel strange, and hard coding every mapping there would make it really hard to maintain. So put it here temporarily until we figure out a better way computing it in `testgrid.go`

Reply:
I dont think its strange. We dont have to put all the mappings. Only the ones we need. So, for now we can have only one entry in the map.